packages/marshal/docs/smallcaps-cheatsheet.md

```diff
@@ -0,0 +1,29 @@
+# Smallcaps Cheatsheet
+
+An example-based summary of the Smallcaps encoding
+
+| Passable value  | OCapn name    | JS example          | JSON encoding        |
+| ----------------|---------------|---------------------|----------------------|
+| bigint          | SignedInteger | `7n`<br>`-7n`       | `"+7"`<br>`"-7"`     |
+| manifest constant | Undefined<br><br>Float64<br><br><br> | `undefined`<br>`Infinity`<br>`-Infinity`<br>`NaN`<br>`-0` | `"#undefined"`<br>`"#Infinity"`<br>`"#-Infinity"`<br>`"#NaN"`<br>`"#-0"` // unimplemented |
+| passable symbol | Symbol        | `Symbol.for('foo')`<br>`Symbol.asyncIterator` | `"%foo"`<br>`"%@@asyncIterator"` |
+| remotable       | Capability    | `Far('foo', {})`    | `"$0.foo"`           |
+| promise         | Capability    | `Promise.resolve()` | `"&1"`               |
+| special string  | String        | `'#foo'`            | `"!#foo"`            |
+| other string    | String        | `'foo'`             | `"foo"`              |
+| other JSON scalar | Null<br>Boolean<br><br>Float64 | `null`<br>`true`<br>`false`<br>`7.1` | `null`<br>`true`<br>`false`<br>`7.1` |
+| copyArray       | Sequence      | `[a,b]`             | `[<a>,<b>]`          |
+| copyRecord      | Struct        | `{x:a,y:b}`         | `{<x>:<a>,<y>:<b>}`  |
+| error           | Error         | `TypeError(msg)`    | `{"#error":<msg>,"name":"TypeError"}` |
+| tagged          | Tagged        | `makeTagged(t,p)`   | `{"#tag":<t>,"payload":<p>}` |
```

Comment:
I recommend also showing the JS example for a tagged that does not rely on the `makeTagged` utility, since this illustrates why tagged values do not overlap with the representations of any other value, and clarifies why we can’t support structs with symbol keys.

Reply:
Based on feedback from the ocapn meeting, and also a private conversation with @cwebber , I'm trying to keep this as small as possible while still conveying enough info that they can get started. (That's why I called it only a "Cheatsheet".)

@cwebber , did I understand what you need correctly? For your (and ocapn's) purposes, would you rather see these details added or omitted? It would certainly be little trouble to add. Likewise for @kriskowal 's suggestions below.

Reply:
> clarifies why we can’t support structs with symbol keys.

Added a bullet point stating that Structs can only have string keys.

Reply:
@kriskowal @cwebber , with the added bullet points, I feel close enough to @kriskowal 's suggestions to make use of his approval to merge this, which I will proceed to do.

@kriskowal,  @cwebber, other reviewers, I'd still like feedback. It is simple enough to do post-merge adjustments on this doc.